src/sage/schemes/elliptic_curves/ell_generic.py

```diff
@@ -2352,126 +2389,51 @@ def multiplication_by_m(self, m, x_only=False):
                 return x
 
         # Grab curve invariants
-        a1, a2, a3, a4, a6 = self.a_invariants()
+        a1, _, a3, _, _ = self.a_invariants()
 
         if m == -1:
             if not x_only:
                 return (x, -y-a1*x-a3)
             else:
                 return x
 
-        # the x-coordinate does not depend on the sign of m.  The work
+        # If we only require the x coordinate, it is faster to use the recursive formula
+        # since substituting polynomials is quite slow.
+        p = Integer(self.base_ring().characteristic())
+        v_p = 0 if p == 0 else valuation(m.abs(), p)
```

Comment:
The `valuation()` is invariant under multiplication by units, so taking the `.abs()` is unnecessary.